Sources/BowFree/Coyoneda.swift

```diff
@@ -2,66 +2,157 @@ import Foundation
 import Bow
 
 public final class ForCoyoneda {}
-public typealias AnyFunc = (AnyObject) -> AnyObject
-public final class CoyonedaPartial<F, P>: Kind2<ForCoyoneda, F, P> {}
-public typealias CoyonedaOf<F, P, A> = Kind<CoyonedaPartial<F, P>, A>
+public final class CoyonedaPartial<F>: Kind<ForCoyoneda, F> {}
+public typealias CoyonedaOf<F, A> = Kind<CoyonedaPartial<F>, A>
 
-public class Coyoneda<F, P, A>: CoyonedaOf<F, P, A> {
-    fileprivate let pivot: Kind<F, P>
-    fileprivate let ks: [AnyFunc]
-
-    public static func apply(_ fp : Kind<F, P>, _ f : @escaping (P) -> A) -> Coyoneda<F, P, A> {
-        return unsafeApply(fp, [f as! AnyFunc])
-    }
-
-    public static func unsafeApply(_ fp : Kind<F, P>, _ fs : [AnyFunc]) -> Coyoneda<F, P, A> {
-        return Coyoneda<F, P, A>(fp, fs)
+public class Coyoneda<F, A>: CoyonedaOf<F, A> {
```

Comment:
Can't Coyoneda be final?

Reply:
Yes, I forgot to add the final keyword. The Arbitrary instance is also in place now.